### PR TITLE
[AMS] Refactor

### PR DESCRIPTION
**Pull request by mzegla**
* introduced ModelBuilder class
* moved configuration loading logic to ModelBuilder
* added top level config schema validation
* moved configuration details from config.py to models json files
* changed ovms_connector so it could include multiple inputs in the request
* adjusted unit tests

### CODE REVIEW

File: extras/ams_wrapper/src/config.py [hunk 1]
```diff
@@ -20,62 +20,7 @@
...
-    "class": DetectionModel, 
-    "ovms_mapping": {
-        "model_name": "face_detection_adas",
-        "model_version": DEFAULT_VERSION,
-        "input_name": "data",
-        },
-    "config_path": "/opt/ams_models/face_detection_adas.json"
-    },
-]
+TYPE_CLASS_MAPPING = {
```
Comment (rasapala) [inline extras/ams_wrapper/src/config.py:23]: Maybe we can move this definition elsewhere to remove the config.py file ? wrapper.py ?

File: extras/ams_models/person_vehicle_bike_detection_model.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
-    "model_name": "person-vehicle-bike-detection-crossroad",
+    "model_name": "personVehicleBikeDetection",
```
Comment (rasapala) [inline extras/ams_models/person_vehicle_bike_detection_model.json:2]: Should we change this property to endpoint_name ?

File: extras/ams_wrapper/src/config.py (hunk 1 above)
Comment (mzegla) [reply extras/ams_wrapper/src/config.py:23]: How about just renaming it? I think it would be good to have all AMS related constants in one place.

Comment (rasapala) [reply extras/ams_wrapper/src/config.py:23]: Ok

File: extras/ams_wrapper/src/api/models/model_config.py [hunk 3]
```diff
@@ -15,6 +15,13 @@
 #
 
 from marshmallow import Schema, fields, validates_schema, ValidationError, post_load, validate
+    
+class ModelConfigurationSchema(Schema):
+    model_name = fields.String(required=True)
+    model_type = fields.String(required=True)
+    inputs = fields.List(fields.Dict(keys=fields.String(), required=True))
+    outputs = fields.List(fields.Dict(keys=fields.String(), required=True))
```
Comment (mateusz-ciesielski) [inline extras/ams_wrapper/src/api/models/model_config.py:23]: Shouldn't it be: 
```python
inputs = fields.List(fields.Nested(ModelInputConfigurationSchema, required=True))
outputs = fields.List(fields.Nested(ModelOutputConfigurationSchema, required=True))
```
?

**Pull request merged**